packages/headless/src/features/folding/folding-slice.ts

```diff
@@ -33,70 +40,141 @@ function areDefinedAndEqual<T>(
   return (value1 || value2) !== undefined && value1 === value2;
 }
 
-function flattenResultTree(result: ResultWithFolding): ResultWithFolding[] {
-  const results = [result];
-  if (result.parentResult) {
-    results.unshift(result.parentResult);
-  }
-  if (result.childResults?.length) {
-    results.push(
-      ...result.childResults.flatMap((childResult) =>
-        flattenResultTree(childResult)
-      )
-    );
+function getChildResultsRecursively(
```

Comment:
This function and the others below in this file were basically rewritten, the diff won't be of any use.